### PR TITLE
feat(sdk-go): enforce free-text ceilings from the schema package

### LINKED ISSUE #503: sdk-go: enforce insight/free-text maxLength from the schema package
Part of #501. Depends on the schema sub-issue in #501.

Make the Go SDK input types reject over-limit free-text fields, using the ceilings exposed by the schema package rather than hardcoded literals.

## Scope

- Enforce the ceilings on the Go input types for the fields in the #501 ceiling table, sourcing each limit from the schema package accessors (the SDK already imports the schema package for scoring — see `sdk/go/scoring.go`).
- Reject over-limit input with a clean, domain-worded error naming the field and its limit; never truncate.
- Add table-driven tests: at-limit accepted, one-over rejected, per field.

## Acceptance criteria

- [ ] No hardcoded limit literal in `sdk/go`; each comes from the schema accessor.
- [ ] At-limit / over-limit tests pass for every bounded field.
- [ ] `make lint-sdk-go` / `make test-sdk-go` green.

### PR DESCRIPTION
**Pull request by peteski22**
## Summary

Enforces the schema's free-text length ceilings and array-cardinality bounds on the Go SDK's input surface, sourcing every limit from the `cqschema` package accessors (no hardcoded literals), so the SDK rejects oversized input before it reaches the wire.

This is the Go counterpart of the merged Python SDK PR #510, keeping the two SDKs in parity: a unit one SDK accepts is one the other accepts.

Per the RFC's implementer guidance (#500), rejection is actionable — the error names the field, its limit, and the actual length — so an LLM producer can re-emit a shorter unit rather than getting an opaque failure.
Nothing is ever truncated.

## Changes

- `validate.go` (new) — `validateProposeParams` checks every free-text field (summary/detail/action/created_by/pattern) and tag item (domains/languages/frameworks) against its schema `maxLength`, plus the domains/languages/frameworks cardinality bounds, collecting all violations via `errors.Join`.
  Length is counted in Unicode code points (`utf8.RuneCountInString`) to match the schema's `maxLength` semantics, so multibyte input is measured the same way here as at the wire boundary.
- `client.go` — `Propose` rejects over-limit params before building or storing the unit; `Flag` rejects an over-limit `WithDetail`.
- `validate_test.go` (new) — table-driven at-limit/over-limit tests per field (length and cardinality), a runes-not-bytes guard, and boundary tests asserting the exact error string a `Propose`/`Flag` caller receives and that a rejected propose stores nothing.
- `go.mod`/`go.sum` (and the postgres store submodule) — bump `github.com/mozilla-ai/cq/schema` to v0.3.1, the release that exposes the ceiling accessors.

## Verification

- `make test-sdk-go` — all modules pass
- `make lint-sdk-go` — golangci-lint 0 issues
- Full `make lint` / `make test` — green

Closes #503

## Summary by CodeRabbit

* **Bug Fixes**
  * Added validation for proposal fields, tags, item counts and flag details.
  * Over-limit input is now rejected with clear errors rather than being truncated or stored.
  * Length checks correctly account for Unicode characters.

* **Tests**
  * Added coverage for valid boundary values, rejected over-limit values and prevention of persistence after failed proposals.

### CODE REVIEW

File: sdk/go/validate.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		validateLength("detail", params.Detail, cqschema.DetailMaxLength()),
+		validateLength("action", params.Action, cqschema.ActionMaxLength()),
+		validateLength("created_by", params.CreatedBy, cqschema.CreatedByMaxLength()),
+		validateLength("pattern", params.Pattern, cqschema.PatternMaxLength()),
+		validateItemLengths("domain", params.Domains, cqschema.DomainMaxLength()),
+		validateItemLengths("language", params.Languages, cqschema.LanguageMaxLength()),
+		validateItemLengths("framework", params.Frameworks, cqschema.FrameworkMaxLength()),
+		validateCount("domains", params.Domains, cqschema.DomainsMaxItems()),
+		validateCount("languages", params.Languages, cqschema.LanguagesMaxItems()),
+		validateCount("frameworks", params.Frameworks, cqschema.FrameworksMaxItems()),
```
[the remark this answers is not in the document]
Comment (peteski22) [reply sdk/go/validate.go:58]: Correct that `validateProposeParams` doesn't check `minItems: 1` here — keeping it out of this PR for two reasons:

- **Scope:** this PR (and the #501 epic) enforce the *ceilings* — `maxLength`/`maxItems` upper bounds. `minItems: 1` is a lower bound / required-ness constraint the epic doesn't cover.
- **Already enforced:** an empty `Domains` slice is rejected inside the SDK at the store layer (`store_memory.go`, `store_sqlite.go`: `"knowledge unit must have at least one non-empty domain"`), so `Propose` already returns an error for it rather than silently accepting it — the local path via `store.Insert`, the remote path via the remote's 422.

The valid consistency point — enforcing required-ness at the same `Propose` boundary as the ceilings, for parity with the Python model — is tracked in #515.
